Add nullptr check in maybeExtractNearestSourceLoc

augusto2112 opened the pull request:
rdar://127251788

augusto2112 commented:
@swift-ci smoke test

adrian-prantl commented:
I don't understand how the test failure only happens on this branch, but: https://github.com/apple/llvm-project/pull/8714

augusto2112 commented:
@swift-ci smoke test

xedin commented on lib/AST/DeclContext.cpp:
```diff
@@ -1418,6 +1418,9 @@ bool DeclContext::isAsyncContext() const {
 
 SourceLoc swift::extractNearestSourceLoc(const DeclContext *dc) {
+  if (!dc)
+    return SourceLoc();
```
This feels like it should be an assert instead and callers should be responsible for action on `nullptr`. It should be possible to add additional checks to `maybeExtractNearestSourceLoc` in `SimpleRequest` to prevent calls to `extractNearestSourceLoc` on invalid inputs.

augusto2112 replied:
@xedin I added the check in `maybeExtractNearestSourceLoc` for any pointer types instead, let me know what you think

xedin replied:
Looks good! Please add parenthesis though before merging.

augusto2112 replied:
@xedin, sorry, parenthesis around what? The constexpr?

xedin replied:
Sorry, I meant curly braces 🤦 At least for the outer if statement with constexpr, ideally both.

augusto2112 committed:
Add nullptr check in maybeExtractNearestSourceLoc

For templated types which are pointers, check if the pointer is null in
maybeExtractNearestSourceLoc.

rdar://127251788

augusto2112 commented:
@swift-ci smoke test

augusto2112 commented:
@swift-ci smoke test
